apps/prairielearn/src/pages/partials/groupRoleTable.ejs

```diff
@@ -0,0 +1,81 @@
+<% const roleConfigProblems = (rolesInfo.validationErrors?.length ?? 0) + (rolesInfo.usersWithoutRoles.length > 0 ? 1 : 0) + (rolesInfo.rolesAreBalanced ? 0 : 1); %>
```

Comment:
Nit, this should be equivalent and more consistent with the handling of `rolesInfo.validationErrors`?

From what I can tell in the TS types, we always expect `validationErrors` to be present. Is the optional chaining on it to get `length` necessary?

```suggestion
<% const roleConfigProblems = (rolesInfo.validationErrors?.length ?? 0) + rolesInfo.usersWithoutRoles.length + (rolesInfo.rolesAreBalanced ? 0 : 1); %>
```

Reply:
Indeed, if `rolesInfo` is not null, then `validationErrors` is not null. I'll remove the optional chaining.

As for the `usersWithoutRoles.length`, this comes from #8006, and is based on the fact that a single alert is created in this case. So it's a count of alerts, not a count of problems. Do you want to change this?

Reply:
Ahh, I didn't read far enough down. I thought this was just for the conditional on the next line, but I see that we actually render a badge with this count. You can keep the count as it.